### ISSUE TITLE
🧪 [Experiment] MarqueeTextBlock

### ISSUE THREAD

**Issue by niels9001** [CommunityToolkit/Labs-Windows#426]
### Approved from Discussion

https://github.com/CommunityToolkit/Labs-Windows/discussions/231

### Problem Statement

At times text might not fit the screen - or you'd want to have a specific experience where text can be animated / looped, similar to what you'd commonly find in newsfeeds.

### Overview

This experiment adds the following components:
 - **MarqueeTextBlock**

### Using

You can try it out via the NuGet Packages here:
 - UWP: https://dev.azure.com/dotnet/CommunityToolkit/_artifacts/feed/CommunityToolkit-Labs/NuGet/CommunityToolkit.Labs.Uwp.MarqueeText
 - WinUI 3: https://dev.azure.com/dotnet/CommunityToolkit/_artifacts/feed/CommunityToolkit-Labs/NuGet/CommunityToolkit.Labs.WinUI.MarqueeText
Read more about [Preview Packages here](https://aka.ms/wct/wiki/previewpackages).

**Code**
```xml
 <labs:MarqueeText Behavior="Ticker"
                          Direction="Left"
                          FontSize="18"
                          RepeatBehavior="Forever"
                          Speed="{96"
                          Text="Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua." />
```

**Result**

**Documentation & samples**
https://github.com/CommunityToolkit/Labs-Windows/tree/main/components/MarqueeText/samples

### Implementation Requirements

- [X] Working Prototype
- [X] Feature Complete
- [X] Documentation
- [X] Samples
- [ ] Unit Tests
- [ ] Community Feedback / Usage Testimonies

### Tested Platforms

- [X] UWP
- [X] WinAppSDK / WinUI 3
- [X] Web Assembly (WASM)
- [ ] Android
- [ ] iOS
- [ ] MacOS
- [ ] Linux / GTK

### Technical Review

- [ ] Accessibility Audit
- [ ] API/Naming Review
- [ ] Code Quality/Style
- [ ] Dependency Review
- [ ] Design/Style Review
- [ ] Final Approval

### Community Help?

None

**Comment by Jay-o-Way** [comment]
Semantics: _forever_ versus _infinite_.

**Comment by Lightczx** [comment]
@niels9001 Hi, I'm using this control in my project currently,
I want to implement a feature that when the pointer entered the text start animation,
and when pointer exited, the text should be re-placed at the very begin place.
is there any way to programmatically achieve this? Or should this be a feature request?

**Comment by michael-hawker** [comment]
Hi @Lightczx, sorry just saw your message when digging this up for someone else.

Since there's a `Start`/`Stop` method on the control, you could attach a XAML Behavior/EventTrigger to the PointerEntered/Exited and then call those methods. It should be possible without needing to bake anything into the control itself.

**Comment by Lightczx** [comment]
@michael-hawker Thanks for your response, is there any way to controls the initial state?
I prefer not activate it when loaded.

**Comment by Arlodotexe** [comment]
> Hi @Lightczx, sorry just saw your message when digging this up for someone else.
> 
> Since there's a `Start`/`Stop` method on the control, you could attach a XAML Behavior/EventTrigger to the PointerEntered/Exited and then call those methods. It should be possible without needing to bake anything into the control itself.

@Avid29 should be able to shed light on this. I'm checking the code, and this functionality appears to be built-in via the `UpdateAnimation` method. The first method parameter, `resume`, is described as "True if animation should resume from its current position, false if it should restart.", but the method is not public, nor can the value be changed indirectly via the private calls to `UpdateAnimation`. 

@Avid29 When designing this, did you plan for the ability to pause/resume the Marquee, or to change its starting position?

**Comment by Avid29** [comment]
The `resume` parameter is for when the speed or control size changes while the animation is running. It is not supposed to be used for pausing or resuming the animation. 

In design I intended for the Marquee to not play until `StartMarquee` is called. This is not the behavior because of line 56 is `MarqueeText.Events` [link](https://github.com/CommunityToolkit/Labs-Windows/blob/4cc0557ac63a40c672b197cac805eafcc356e534/components/MarqueeText/src/MarqueeText.Events.cs#L56), which automatically begins the animation when the container size is changed. I should have left this behavior up to the user.

Should I release a patch? It's a technically a small breaking change.

**Comment by dishwasher-detergent** [comment]
Hey all, really liking the work that's been done here. We were looking at using this in a project, but the main stopper here is we need the text to be continuous. Currently it waits for the end of the text to go out of view before it starts the loop, we were needing something that would never show that gap and have continuous text scrolling across the screen. 

Just throwing this out there as an idea for the control :)

**Comment by Avid29** [comment]
@dishwasher-detergent This functionality was considered, but I wasn't sure how to make a good interface for it. Something I may do is add an event for when the end of the text is displayed, that way a wrapping control could be made that queues up another piece of text to run in a second `MarqueeText` control after the first triggers that event.

I think that wrapper would be for you to write if you need it though, at least for a bit. An additional control like that is a harder pitch.

**Comment by dishwasher-detergent** [comment]
@Avid29 I think that sounds great, and I'd love to see it in the control. :)

**Comment by TheJoeFin** [comment]
I'm using the this MarqueeText control in my app [trdo](https://github.com/TheJoeFin/Trdo) and would like to add a new mode of 'overflow' which works like marquee text would on a car radio. Only marquee the text when it needs to and wait a couple of seconds before doing the marquee. Are folks interested in this addition?

https://github.com/user-attachments/assets/10a9443d-adfb-47d1-ad75-4b6800661d9a

**Comment by Avid29** [comment]
@TheJoeFin The control is already designed with this use case in mind.

This behavior can be achieved using the `Looping` mode and periodically calling `StartMarquee` either in code-behind or using XAML Behaviors.

**Comment by TheJoeFin** [comment]
@Avid29 you're right! I have created this behavior in my app Trdo now using the existing MarqueeText control. There are obviously more steps with event subscriptions and timers, but it works well!

https://github.com/user-attachments/assets/1d54149d-844b-4b4d-8142-89f45361112b
